sklearn/utils/estimator_checks.py

```diff
@@ -2384,6 +2385,71 @@ def check_classifiers_regression_target(name, estimator_orig):
         assert_raises_regex(ValueError, msg, e.fit, X, y)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_estimator_sparse_dense(name, estimator_orig):
+    rng = np.random.RandomState(52)
+    estimator = clone(estimator_orig)
+    estimator_sp = clone(estimator_orig)
+    tags = _safe_tags(estimator_orig)
+    centers = 2 if tags["binary_only"] else None
+    X, y = make_blobs(random_state=rng, cluster_std=0.5, centers=centers)
+    # for put some points to zero to have a little bit of sparsity
+    X -= np.min(X) - 1.  # we need positive data for ComplementNB for instance
```

Comment:
This could/should use the `requires_positive_X` estimator tag.

Reply:
Agreed, I did the replacement now, although it will work only when #14680 is merged